### PR TITLE
fix(plugin-server): Fix cohort matching in actions

### LINKED ISSUE #11389: Cohort matching logic in plugin-server is incorrect
## Bug description

In https://github.com/PostHog/posthog/pull/8449 @EDsCODE introduced versioning to cohort people.

This logic is not handled correctly in plugin-server/src/utils/db/db.ts#addPersonToCohort

## Problems

1. We should not be querying clickhouse
2. `posthog_cohortpeople` query should also check against latest version of the cohort

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

## Additional context

cc @neilkakkar @yakkomajuri

### PR DESCRIPTION
**Pull request by macobo**
Fixes https://github.com/PostHog/posthog/issues/11389 by updating the relevant query and removes dependency on clickhouse

Verified that the new query is doing index-only scans as well :)

### CODE REVIEW

File: plugin-server/tests/main/db.test.ts [hunk 1]
```diff
@@ -1095,4 +1095,44 @@ describe('DB', () => {
...
+        let cohort: Cohort
+        let person: Person
+
+        beforeEach(async () => {
+            team = await getFirstTeam(hub)
+            cohort = await hub.db.createCohort({
+                name: 'testCohort',
+                description: '',
+                team_id: team.id,
+                version: 10,
```
Comment (neilkakkar) [inline plugin-server/tests/main/db.test.ts:1110]: can we have another test where version is not set, to simulate static cohorts?

**Review by neilkakkar — APPROVED** [review]

File: plugin-server/tests/main/db.test.ts (hunk 1 above)
Comment (macobo) [reply plugin-server/tests/main/db.test.ts:1110]: What is the value then? Null? Something else?

Comment (macobo) [reply plugin-server/tests/main/db.test.ts:1110]: Looking at production, seems like version is still set for static cohorts: https://metabase.posthog.net/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjpudWxsLCJuYXRpdmUiOnsicXVlcnkiOiIiLCJ0ZW1wbGF0ZS10YWdzIjp7fX0sInR5cGUiOiJuYXRpdmUifSwiZGlzcGxheSI6InRhYmxlIiwidmlzdWFsaXphdGlvbl9zZXR0aW5ncyI6e319

Comment (neilkakkar) [reply plugin-server/tests/main/db.test.ts:1110]: refresh the link? Shows up empty for me

Comment (macobo) [reply plugin-server/tests/main/db.test.ts:1110]: The query was `SELECT * FROM posthog_cohort WHERE version IS NULL`. But in any case, added the test + fixed handling of nulls. Thanks for pointing this out!

File: plugin-server/tests/main/db.test.ts [hunk 2]
```diff
@@ -1095,4 +1095,57 @@ describe('DB', () => {
...
+        })
+
+        it('handles NULL version cohorts', async () => {
+            const cohort2 = await hub.db.createCohort({
+                name: 'null_cohort',
+                description: '',
+                team_id: team.id,
+                version: null,
+            })
+            // expect(await hub.db.doesPersonBelongToCohort(cohort2.id, person)).toEqual(false)
```
Comment (neilkakkar) [inline plugin-server/tests/main/db.test.ts:1145]: debug?

**Review by neilkakkar — APPROVED** [review]
LGTM

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**